jiant/models.py

```diff
@@ -687,6 +692,30 @@ def build_single_sentence_module(task, d_inp: int, project_before_pooling: bool,
     return module
 
 
+def build_sop(task, d_inp, model, params, args):
+    """
+    Build and load the pretrained head for the sentence order prediction task.
+    Right now, there is only support for ALBERT.
+    Parameters
+    ----------
+    task: Task,
+    d_inp: int,
+    model: MultiTaskModel,
+    params: Params
+
+    Returns
+    -------
+    module: Pooler object, which is loaded with pretrained weights from ALBERT SOP
+    pretraining. 
+
+    """
+    module = SOPClassifier(d_inp, task.n_classes, params)
+    # The huggingface implementation exposes the pretrained projection layer for the SOP task, which
+    # we use. See: https://github.com/huggingface/transformers/issues/2671 for more details.
+    module.pooler.project = model.sent_encoder._text_field_embedder.model.pooler
```

Comment:
Note: ALBERT does do pooling the way we do (pooling type = final): https://github.com/google-research/albert/blob/a41cf11700c1ed2b7beab0a2649817fa52c8d6e1/modeling.py#L261

Reply:
I still don't really understand what this is. Are these the final classifier layer parameters (d_hid x 2), or are those loaded somewhere else?

Reply:
Reading the rest of the PR, it doesn't look like those parameters are loaded anywhere, so this shouldn't work... am I missing something?

Reply:
They are used here: https://github.com/nyu-mll/jiant/blob/master/jiant/modules/simple_modules.py#L34 , which is called in the sop_forward (from the SOP Classifier)